server/index.js

```diff
@@ -0,0 +1,37 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const Rooms = require('../database/Room.js');
+
+const app = express()
+const PORT = 3000;
+
+app.use(express.json())
+
+app.get('/api/reviews/:roomid/', (req, res) => {
+  Rooms.find({id: req.params.roomid}, (err, room) => {
+    if (err) {
+      res.sendStatus(500);
+    } else {
+      //the user is able to request a sub-section of room reviews using a query param stylized as such: '?reviewgroup='.
+      var selReviewGroup = Number(req.query.reviewgroup);
+      var maxNumOfReviewsInAGroup = 5;
+      var selReviewGroupIsWithinBounds = (selReviewGroup + 1) * maxNumOfReviewsInAGroup <= room[0].reviews.length;
+
+      //if they did provide a query param and it's within bounds
+      if (selReviewGroup !== undefined && selReviewGroupIsWithinBounds) {
```

Comment:
Currently this check may run into issues with an undefined req.query.reviewgroup.
Have you tried sending a request to the api without a review group yet?

`var selReviewGroup = Number(undefined); // NaN`
`selReviewGroup !== undefined` // True. This will accidentally pass your check`

Reply:
Also see the comment at the bottom for a suggestion to circumvent this problem entirely